### ISSUE TITLE
Menu Addition of public vars from Entity Components

### ISSUE THREAD

**Issue by bturner1273** [AlmasB/FXGL#615]
the new menu is awesome! The only useful feature I can think to include with it is maybe adding public variables from EntityComponents if that is possible.

like in the menu you could have:

Entity Name (dropdown)
-> components (dropdown in entity drop down)
->public vars list for specific component in component drop down

Maybe even let the coder play around with the values? This could be totally not feasible and I would get that...

**Comment by AlmasB** [comment]
This has been added to the latest SNAPSHOT, please can you check

**Comment by bturner1273** [comment]
Just force updated the snapshot, I changed the access modifiers of my variables in my custom components to public but I'm not seeing them show up in the entity menu, the component names do show up however.

**Comment by bturner1273** [comment]
the game vars in the map in initGameVars work perfect

**Comment by AlmasB** [comment]
Sorry, forgot to add that only Observable properties are supported at this moment. So:

```
public class SomeComponent extends Component {
    private DoubleProperty hp = new SimpleDoubleProperty();

    public DoubleProperty hpProperty() {
        return hp;
    }
}
```

It searches all methods, rather than variables, of any component whose names end with `Property` and then binds to them. The reason being:
* we don't want to make variables public and
* not observable values cannot be bound to.

**Comment by bturner1273** [comment]
just changed up a few of my components and it works like a charm. Great addition! Now we just gotta be able to change the values and it would be even that much more powerful

**Comment by AlmasB** [comment]
Merged with #645
